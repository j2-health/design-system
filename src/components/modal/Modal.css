/* Modal Component Styling */
.j2-modal .ant-modal-content {
  background: var(--j2-color-bg-elevated);
  border-radius: var(--j2-border-radius-lg);
  padding: 0;
  overflow: hidden;
}

/* Header styling */
.j2-modal .ant-modal-header {
  background: transparent;
  border-bottom: none;
  padding: var(--j2-padding-md) var(--j2-padding-content-horizontal-lg)
    var(--j2-padding-xs) var(--j2-padding-content-horizontal-lg);
  margin: 0;
  box-shadow:
    0px 1px 2px 0px rgba(0, 0, 0, 0.03),
    0px 1px 6px -1px rgba(0, 0, 0, 0.02),
    0px 2px 4px 0px rgba(0, 0, 0, 0.02);
}

.j2-modal .ant-modal-title {
  font-family: var(--j2-font-family);
  font-weight: var(--j2-font-weight-strong);
  font-size: var(--j2-font-size-lg);
  line-height: var(--j2-line-height-lg);
  color: var(--j2-color-text-heading);
  margin: 0;
}

/* Icon modal specific header styling */
.j2-modal-with-icon .ant-modal-header {
  padding: var(--j2-padding-md) var(--j2-padding-content-horizontal-lg) 0
    var(--j2-padding-content-horizontal-lg);
}

.modal-title-with-icon {
  display: flex;
  gap: var(--j2-padding-sm);
  align-items: flex-start;
}

.modal-icon {
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.modal-title-content {
  flex: 1;
  min-width: 0;
}

.modal-title-text {
  font-family: var(--j2-font-family);
  font-weight: var(--j2-font-weight-strong);
  font-size: var(--j2-font-size-lg);
  line-height: var(--j2-line-height-lg);
  color: var(--j2-color-text-heading);
  margin: 0;
}

/* Close button styling */
.j2-modal .ant-modal-close {
  top: var(--j2-padding);
  right: var(--j2-padding-md);
  width: 22px;
  height: 22px;
  color: var(--j2-color-icon);
}

.j2-modal .ant-modal-close-x {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  font-size: 16px;
  color: var(--j2-color-icon);
}

/* Body styling */
.j2-modal .ant-modal-body {
  padding: 0 var(--j2-padding-content-horizontal-lg) var(--j2-padding-sm)
    var(--j2-padding-content-horizontal-lg);
  font-family: var(--j2-font-family);
  font-weight: var(--j2-font-weight-normal);
  font-size: var(--j2-font-size);
  line-height: var(--j2-line-height);
  color: var(--j2-color-text);
}

/* Icon modal specific body styling */
.j2-modal-with-icon .ant-modal-body {
  padding: var(--j2-padding-xs) var(--j2-padding-content-horizontal-lg)
    var(--j2-padding-sm) var(--j2-padding-content-horizontal-lg);
}

.modal-content-with-icon {
  margin-left: calc(24px + var(--j2-padding-sm));
}

/* Footer styling */
.j2-modal .ant-modal-footer {
  background: transparent;
  border-top: none;
  padding: 0 var(--j2-padding-content-horizontal-lg) var(--j2-padding-md)
    var(--j2-padding-content-horizontal-lg);
  margin: 0;
  text-align: right;
}

.modal-footer {
  display: flex;
  gap: var(--j2-margin-xs);
  justify-content: flex-end;
  align-items: center;
}

.modal-footer-with-icon {
  margin-left: calc(24px + var(--j2-padding-sm));
}

/* Backdrop/mask styling */
.j2-modal .ant-modal-mask {
  background: var(--j2-color-bg-mask);
}
