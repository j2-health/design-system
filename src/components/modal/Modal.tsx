import { Modal as AntdModal, ModalProps } from 'antd'
import {
  XIcon,
  CheckCircleIcon,
  InfoIcon,
  WarningCircleIcon,
  XCircleIcon,
} from '@phosphor-icons/react'
import { Button } from '../button'
import './Modal.css'
import { useMemo } from 'react'
import cx from 'classnames'

type ModalType = 'default' | 'success' | 'info' | 'warning' | 'error'

export type Props = {
  isOpen: boolean
  onClose?: () => void
  onCancel?: () => void
  children: React.ReactNode
  title?: string
  cancelText?: string
  okText?: string
  onOk?: () => void
  type?: ModalType
} & Omit<ModalProps, 'open' | 'onCancel' | 'children'>

const getIcon = (type: ModalType) => {
  switch (type) {
    case 'error':
      return (
        <XCircleIcon color="var(--j2-color-error)" weight="fill" size={24} />
      )
    case 'success':
      return (
        <CheckCircleIcon
          color="var(--j2-color-success)"
          weight="fill"
          size={24}
        />
      )
    case 'info':
      return <InfoIcon color="var(--j2-color-info)" weight="fill" size={24} />
    case 'warning':
      return (
        <WarningCircleIcon
          color="var(--j2-color-warning)"
          weight="fill"
          size={24}
        />
      )
    case 'default':
    default:
      return null
  }
}

const Modal = ({
  isOpen,
  onClose,
  children,
  title = 'Modal title',
  onCancel,
  cancelText = 'Cancel',
  okText = 'Ok',
  onOk,
  type = 'default',
  ...props
}: Props) => {
  const icon = useMemo(() => getIcon(type), [type])

  return (
    <AntdModal
      {...props}
      open={isOpen}
      onCancel={onCancel}
      closable={!!onClose}
      afterClose={() => {
        onClose?.()
      }}
      title={
        icon ? (
          <div className="modal-title-with-icon">
            <div className="modal-icon">{icon}</div>
            <div className="modal-title-content">
              <div className="modal-title-text">{title}</div>
            </div>
          </div>
        ) : (
          title
        )
      }
      footer={
        onCancel || onOk ? (
          <div
            className={cx('modal-footer', {
              'modal-footer-with-icon': icon,
            })}
          >
            {onCancel && (
              <Button type="default" onClick={onCancel}>
                {cancelText}
              </Button>
            )}
            {onOk && (
              <Button type="primary" onClick={onOk}>
                {okText}
              </Button>
            )}
          </div>
        ) : null
      }
      className={cx('j2-modal', { 'j2-modal-with-icon': icon })}
      maskClosable={true}
      centered
      closeIcon={onClose ? <XIcon size={22} weight="regular" /> : null}
    >
      <div
        className={cx('modal-content', {
          'modal-content-with-icon': icon,
        })}
      >
        {children}
      </div>
    </AntdModal>
  )
}

export { Modal }
