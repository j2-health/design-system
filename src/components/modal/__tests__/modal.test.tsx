import { render } from '@testing-library/react'
import { Modal } from '../Modal'

describe('Modal', () => {
  it('should render correctly with onClose', () => {
    const { container } = render(
      <Modal
        isOpen={true}
        onClose={jest.fn()}
        onCancel={jest.fn()}
        onOk={jest.fn()}
      >
        Test content
      </Modal>
    )
    expect(container).toMatchSnapshot()
  })
})
