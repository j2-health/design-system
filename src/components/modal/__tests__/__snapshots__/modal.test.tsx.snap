// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly with onClose 1`] = `<div />`;
